proposals/scopes.md

```diff
@@ -196,106 +196,179 @@ interface OriginalPosition {
+corresponding to the immediately surrounding start/end pair.
+
+```
+generated_range_start :=
+  'F'          // Tag: 0x5 unsigned
+  uFLAGS
+  uLINE?       // Present if FLAGS<0> is set.
+  uCOLUMN
+  sDEFINITION? // Present if FLAGS<1> is set.
+
+generated_range_end :=
+  'G'    // Tag: 0x6 unsigned
+  uLINE?
+  uCOLUMN
+```
+
+Since bundles tend to consist of a single line (or very few lines), `generated_range_start` and `generated_range_end` omit the line by default.
+
+The `uFLAGS` field in `generated_range_start` is a bit field defined as follows:
+  * 0x1: has line
+  * 0x2: has definition
+  * 0x4: is stack frame
+  * 0x8: is hidden
+
+Similar to "original scopes", we use relative numbers to reduce the bytes required:
+
+  * `uLINE` in `generated_range_start` and `generated_range_end` are relative to the previous occurrence (or absolute for the first).
+  * `uCOLUMN` in `generated_range_start` and `generated_range_end` are relative to the previous occurrence, if the previous start/end item is on the same line. Absolute otherwise.
+  * `sDEFINITION` in `generated_range_start` is relative to the previous occurrence (or absolute for the first).
+
+`sDEFINITION` is an index into the list of `original_scope_start` items. If `definitionIdx` is the resolved value, then the corresponding `original_scope_start` could
+be found with the pseudo code `const scopeStart = scopes.filter(item => item.tag === 'B')[definition]`.
+
+```
+generated_range_callsite :=
+  'J'         // Tag: 0x9 unsigned
+  sSOURCE_IDX
+  sLINE
+  sCOLUMN
+```
+
+If a "generated range" contains a callsite, then the range describes an inlined function body. The inlined function was called at the original position described by this `generated_range_callsite`.
+
+  * `sSOURCE_IDX` in `generated_range_callsite` is relative to the previous occurrence (or absolute for the first).
+  * `sLINE` in `generated_range_callsite` is relative to the previous occurrence, if the previous `generated_range_callsite` was in the same source file. Absolute otherwise.
+  * `SCOLUMN` in `generated_range_callsite` is relative to the previous occurrence, if the previous `generated_range_callsite` was on the same line in the same file. Absolute otherwise.
+
+```
+generated_range_bindings :=
+  'H'       // Tag: 0x7 unsigned
+  sBINDING+
+```
+
+`generated_range_bindings` are only valid for generated ranges that have a `sDEFINITION`. The bindings list must be equal in length as the variable list of the original scope the `sDEFINITION` references. `sBINDING+` is a list of indices into the `"names"` field of the source map JSON. Each binding is a JavaScript expression that, when evaluated, produces the **value** of the corresponding variable.
+
+`sBINDING+` indices are encoded absolute. To signify that a variable is unavailable, use the index `-1`.
+
+```
+generated_range_subrange_binding :=
+  'I'             // Tag: 0x8 unsigned
+  uVARIABLE_INDEX
+  binding_from+
```

Comment:
I think we need to add the number of subrange bindings here.

Reply:
Same reasoning as above for variables.